tiledb/cloud/_common/ordered.py

```diff
@@ -0,0 +1,82 @@
+"""Implementations of Python sets that remember their order."""
+
+from typing import AbstractSet, Iterable, Iterator, MutableSet, TypeVar
+
+_T = TypeVar("_T")
+_T_co = TypeVar("_T_co", covariant=True)
+_Self = TypeVar("_Self", bound="FrozenSet")
+"""Annotation for returning an object of your own type.
+
+See https://peps.python.org/pep-0673/.
+"""
+
+
+class FrozenSet(AbstractSet[_T_co]):
+    """A frozenset that remembers the insertion order of elements.
+
+    Unlike dicts, Python sets don't remember insertion order. This class
```

Comment:
boo! insertion-order-disrespecting sets/dicts are (IMNSHO) a recipe for disaster -- i say that over many years with many languages. one of the things javascript gets right .... :D 

what ends up happening is good solid people like yourself end up inventing what should have been the default all along